src/ucp/core/ucp_request.inl

```diff
@@ -96,6 +116,7 @@
             } \
         } else { \
             __req = ((ucp_request_t*)(_param)->request) - 1; \
+            ucp_request_pre_init(__req); \
```

Comment:
does it mean we add overhead to tag_send with user-defined request? like in MPI_Send ?

Reply:
unfortunately, yes

Reply:
switch implementation to `UCP_REQUEST_FLAG_LOCAL_ID` as it was?
`flags` are already initialized by `0`, so OR-ing `UCP_REQUEST_FLAG_LOCAL_ID` impacts only protocols that allocate local ID

Reply:
i think we need to discuss the option to switch to flags, because of this

Reply:
yes, using flags doesn't add any overhead